src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Controllers/AadSmartOnFhirProxyControllerTests.cs

```diff
@@ -187,5 +189,41 @@ public void GivenMissingAudParamToV1AAd_WhenAuthorizeRequestAction_ThenRedirectR
 
             Assert.ThrowsAsync<AadSmartOnFhirProxyBadRequestException>(() => _controller.Token(grantType, compoundCode, redirectUri, clientId, clientSecret)).Wait();
         }
+
+        [Theory]
+        [InlineData(null, null)]
+        [InlineData("user_impersonation test.scope", "user_impersonation test.scope")]
+        [InlineData("patient$Observation.read", "patient/Observation.read")]
+        [InlineData("patient$Observation.read patient$Encounter.read", "patient/Observation.read patient/Encounter.read")]
+        public void GivenScopesInTokenResponse_WhenTokenRequestAction_ThenCorrectScopesReturned(string inScopes, string outScopes)
```

Comment:
nit: parameter names - probably something like `tokenScopes` and `expectedScopes`